recipe-server/normandy/recipes/api/views.py

```diff
@@ -120,7 +120,13 @@ def history(self, request, pk=None):
     def enable(self, request, pk=None):
         recipe = self.get_object()
         recipe.enabled = True
-        recipe.save()
+
+        try:
+            recipe.save()
+        except Recipe.NotApproved:
+            return Response({'enabled': 'Cannot enable a recipe that has not been approved.'},
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
IMO a 400 is more for malformed requests than for a request that isn't allowed by business logic. Maybe this is a 403, or a 409?